test/api/controller/base.test.js

```diff
@@ -633,9 +634,38 @@ describe('#base/native controller', () => {
       should(searchBody).be.eql({ query: { foo: 'bar' } });
     });
 
+    it('should extract search params', () => {
+      request.context = { connection: { protocol: 'http', misc: { verb: 'GET' } } };
+      request.input.args.query = '{"query":{"foo":"bar"}}';
+      const
+        { from, size, scrollTTL, query, searchBody } = nativeController.getSearchParams(request);
+
+      should(from).be.eql(1);
+      should(size).be.eql(11);
+      should(scrollTTL).be.eql('10m');
+      should(query).be.eql({ foo: 'bar' });
+      should(searchBody).be.eql({ query: { foo: 'bar' } });
+    });
+
+    it('should extract searchBody param', () => {
+      request.context = { connection: { protocol: 'http', misc: { verb: 'GET' } } };
+      request.input.args.query = '{"query":{"foo":"bar"}}';
+      const searchBody = nativeController.getSearchBody(request);
+
+      should(searchBody).be.eql({ query: { foo: 'bar' } });
+    });
+
+    it('should extract searchBody param', () => {
```

Comment:
missing tests on what happens if:
* an invalid search body is provided
* no search body is provided and the route is invoked using HTTP:GET

Reply:
shouldnt we provide an empty searchBody by default since that's what we do when we use POST ?

Reply:
Of course. But tests don't check if that happens.